File: client/src/services/conversations.ts
import { apiUrl } from "@/lib/api";

export async function fetchConversations() {
  const res = await fetch(apiUrl("/api/conversations"), { credentials: "include" });
  if (!res.ok) throw new Error("Failed to fetch conversations");
  return res.json();
}

export async function fetchConversation(id: number) {
  const res = await fetch(apiUrl(`/api/conversations/${id}`), { credentials: "include" });
  if (!res.ok) throw new Error("Conversation not found");
  return res.json();
}

export async function sendMessage(conversationId: number, content: string) {
  const res = await fetch(apiUrl(`/api/conversations/${conversationId}/messages`), {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    credentials: "include",
    body: JSON.stringify({ content }),
  });
  if (!res.ok) throw new Error("Failed to send message");
  return res.json();
}

File: reachly-server/src/db/conversations.rs
import { api, type InsertLead } from "@shared/routes";
import { useToast } from "@/hooks/use-toast";
import { fetchLeads } from "@/services/leads";
import {
  useQuery,
  useMutation,
  useQueryClient,
} from "@tanstack/react-query";

export function useLeads() {
  return useQuery({
    queryKey: ["leads"],
    queryFn: fetchLeads,
  });
}

export function useCreateLead() {
  const queryClient = useQueryClient();
  const { toast } = useToast();

  return useMutation({
    mutationFn: async (data: InsertLead) => {
      const res = await fetch(api.leads.create.path, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(data),
        credentials: "include",
      });

      if (!res.ok) {
        if (res.status === 400) {
          const error = api.leads.create.responses[400].parse(
            await res.json()
          );
          throw new Error(error.message);
        }
        throw new Error("Failed to create lead");
      }

      return api.leads.create.responses[201].parse(await res.json());
    },

    onSuccess: () => {
      // ✅ must match ["leads"]
      queryClient.invalidateQueries({ queryKey: ["leads"] });

      toast({
        title: "Lead Created",
        description: "Successfully added new lead.",
      });
    },

    onError: (error: Error) => {
      toast({
        variant: "destructive",
        title: "Error",
        description: error.message,
      });
    },
  });
}
